python-frontend/src/test/java/org/sonar/python/semantic/v2/TypeInferenceV2Test.java

```diff
@@ -3124,4 +3262,25 @@ private static Statement lastStatement(StatementList statementList) {
     List<Statement> statements = statementList.statements();
     return statements.get(statements.size() - 1);
   }
+
+  private static class TestProject {
```

Comment:
That's a pretty neat construct!
I'm wondering if it wouldn't make more sense for this to be in a different test category though (either `ProjectLevelSymbolTableTest` or `ProjectLevelTypeTableTest`, or even a new `CrossFileTypeInferenceTest` dedicated to tests focused on import resolution.
We probably can keep this here for now though.